backend/auth/firebase.go

```diff
@@ -16,55 +16,92 @@ package auth
 
 import (
 	"context"
-	"errors"
 	"firebase.google.com/go/v4/auth"
 	"net/http"
+	"peoplemath/models"
 	"strings"
 	"time"
 )
 
 type FirebaseAuth struct {
 	FirebaseClient firebaseAuthClient
 	AuthTimeout    time.Duration
-	AuthDomain     string
 }
 
 type firebaseAuthClient interface {
 	VerifyIDToken(ctx context.Context, idToken string) (*auth.Token, error)
 }
 
-func (auth *FirebaseAuth) Authorize(next http.HandlerFunc) http.HandlerFunc {
+func (auth FirebaseAuth) Authenticate(next http.HandlerFunc) http.HandlerFunc {
```

Comment:
Shall we keep this as a pointer receiver?

Reply:
Yes, I don't see any reason against making this change as a precaution.